tests/test_optimizer.py

```diff
@@ -602,6 +602,16 @@ def test_expand_alias_refs(self):
             "WITH data AS (SELECT 1 AS id, 2 AS my_id, 'a' AS name, 'b' AS full_name) SELECT data.id AS my_id, CONCAT(data.id, data.name) AS full_name FROM data WHERE data.id = 1 GROUP BY data.id, CONCAT(data.id, data.name) HAVING data.id = 1",
         )
 
+        # Edge case: BigQuery shouldn't expand aliases that are reusing input column names in expressions
```

Comment:
I think we can improve this a bit, e.g. "BigQuery shouldn't expand aliases in complex expressions" or something. The "that are reusing input column names" is confusing to me.